### PR TITLE
feature/project-persistence-bus-4

### PR DESCRIPTION
**Pull request by Timbogen**

### CODE REVIEW

File: Assembus/Assets/Scripts/Services/ProjectManager.cs [hunk 1]
```diff
@@ -60,32 +92,159 @@ private ProjectManager()
...
-            // Load object model into the MainScene
-            var (parent, children) = ObjectLoader.LoadObject(importPath);
-            if (parent == null)
+        /// <summary>
+        ///     Create the necessary files for the project and copy
+        ///     the import object.
+        /// </summary>
+        /// <param name="projectPath">The path to the project</param>
+        /// <param name="importPath">The path to the import object</param>
+        /// <returns></returns>
```
Comment (gh-nalo) [inline Assembus/Assets/Scripts/Services/ProjectManager.cs]: Missing return description

File: Assembus/Assets/Scripts/Services/ProjectManager.cs [hunk 2]
```diff
@@ -60,32 +92,159 @@ private ProjectManager()
...
+            }
+
+            // Try to save the project
+            return SaveProject();
+        }
 
-            return (true, "");
+        /// <summary>
+        ///     Save all project data from the local ProjectSpace instance to disk
+        /// </summary>
```
Comment (gh-nalo) [inline Assembus/Assets/Scripts/Services/ProjectManager.cs:153]: Missing return description

File: Assembus/Assets/Scripts/Services/ProjectManager.cs [hunk 3]
```diff
@@ -60,32 +92,159 @@ private ProjectManager()
...
+            }
+            catch (Exception)
+            {
+                return (false, "The project couldn't \n be saved!");
+            }
+        }
+
+        /// <summary>
+        ///     Load all project data into the local ProjectSpace instance
+        /// </summary>
```
Comment (gh-nalo) [inline Assembus/Assets/Scripts/Services/ProjectManager.cs:180]: Missing return description

Comment (MaxPeltzer) [reply Assembus/Assets/Scripts/Services/ProjectManager.cs:189]: Fixed

File: Assembus/Assets/Scripts/Services/ProjectManager.cs (hunk 2 above)
Comment (MaxPeltzer) [reply Assembus/Assets/Scripts/Services/ProjectManager.cs:162]: Fixed

File: Assembus/Assets/Scripts/Services/ProjectManager.cs (hunk 1 above)
Comment (MaxPeltzer) [reply Assembus/Assets/Scripts/Services/ProjectManager.cs]: Fixed

**Review by MaxPeltzer — APPROVED** [review]
Fixed all the flaws I found and the ones mentioned by @gh-nalo. 
There is one more open Comment which should be reviewed by someone else and may be open for discussion. In my opinion this problem can be postponed, thus I approve merging these changes to the main branch.

File: Assembus/Assets/Scripts/MainScreen/UIController.cs [hunk 4]
```diff
@@ -55,21 +104,57 @@ private void Update()
             _width = width;
         }
 
+        /// <summary>
+        ///     Set the name of the current project
+        /// </summary>
+        private void OnEnable()
+        {
+            title.text = _projectManager.CurrentProject.Name + "*";
```
[the remark this answers is not in the document]
Comment (pscheede) [reply Assembus/Assets/Scripts/MainScreen/UIController.cs:113]: Yes, this was included to be able to show the unsaved changes until any changes can be saved, and will be changed later.

**Review by pscheede — DISMISSED** [review]

**Review by pscheede — DISMISSED** [review]

**Review by pscheede — APPROVED** [review]

**Review by MaxPeltzer — APPROVED** [review]

**Pull request merged**